### ISSUE TITLE
'Bob: Create' button does nothing in Chrome 71.0.3578.98

### ISSUE THREAD

**Issue by KevCoLaboratories** [xem/miniWebRTC#3]
Hello, Xem!

**PREFACE:**
I've been tinkering with various personal WebRTC-related projects here and there for a couple years and have used GitHub to read how others have overcome certain issues, but this is the first time I've ever actually posted on GitHub, so please forgive any GitHub faux pas I may unintentionally commit.

First, let me thank you from the bottom of my heart for whittling down the serverless-webrtc code into this easy to read, simplified version. It looks extremely promising and takes a lot of the "magic box" guesswork out of the whole process.

---------------------------
**THE ISSUE:**
I'm running Google Chrome Version 71.0.3578.98 (Official Build) (64-bit) on Windows 10, and when I click 'Bob: Create', nothing happens. No errors or messages in the console. No content to copy from the 'localOffer' text input field. Nothing.

---------------------------
**FURTHER TESTING IN CHROME:**
I went on to see if ANY buttons give any reaction in Chrome, just to verify that Chrome didn't completely lock-up or something.

If I click either 'Okay, I pasted it.' buttons, I'm given an "Uncaught SyntaxError: Unexpected end of JSON input" error in the console.

If I type in some text in the 'messageTextBox' text input field and press the 'Send message' button, I'm given a "Uncaught DOMException: Failed to execute 'send' on 'RTCDataChannel': RTCDataChannel.readyState is not 'open'" error in the console.

---------------------------
**TRYING AN ALTERNATIVE BROWSER:**
If I run it in Firefox 63.0.3 (64-bit), and I click 'Bob: Create', an offer appears in the 'localOffer' text input field. I am able to copy/paste this offer and follow the remainder of the steps until I get the error message "ICE failed, add a TURN server and see about:webrtc for more details" in the console that I noticed another user [Atzen2] encountered back in 2017.

---------------------------
**CONCLUSION:**
My main goal is to get this running in Chrome, so the Firefox issue does not really concern me at the moment. Do you think this is a Chrome-version specific issue or am I doing something wrong?

I feel like this is just some tiny little issue which I'm overlooking that's holding up the majority of the code from processing.

I truly appreciate your time and I thank you for any help you're willing to provide.

-Kevin

**Comment by xem** [comment]
Hello and thanks for your message!
Don't worry, there's no "wrong way" to send me feedback, and your issue is very kind and well written! :)

Like you, I searched everywhere for a simple "serverless" webrtc demo, and as nothing fitted my expectations, I took cjb's project and removed every single line of code that wasn't mandatory.
This was back in 2016, this technology was still being standardized (I'm not sure if it is now) and cjb's project, as well as mine, regularly stopped working on some browsers while they released new versions, and broke things in the process.

Though this time, it doesn't seem to be the case.

I just opened https://xem.github.io/miniWebRTC/ on Firefox and Chrome side-by-side on Windows 10. They have no important extensions enabled besides an ad-blocker (which doesn't cause any problem here), and no flags enabled in the options.

Chrome (on the left) was Bob and Firefox (on the right) was Alice.
I managed to play all the scenario and make the two browsers chat with each other.
You can see a screen record of this test here: https://www.useloom.com/share/f31d851b6c0e4cd9a81a93f028b3b658

I'm sorry I can't troubleshoot further, but if you can't reproduce what I did in that video, I'd say the problem comes from either your browser (maybe an add-on blocking WebRTC) or your antivirus/firewall (I have Avast and it doesn't care) or your Internet connection (are you behind a router, a firewall, a VPN,  a bridge, an auth page, or anything that may block WebRTC?)

I hope it helped (even though I had nothing I could fix in my code). Keep me posted!
Cheers,
Xem

**Comment by KevCoLaboratories** [comment]
Thank you for the extremely prompt response!

So, I disabled all of my extensions to try it again, no luck. 

I logged into Chrome using a different user ("Person") profile (I have several for work), BOOM, it worked flawlessly.

This led me to clear 'Cookies and other site data', 'Cached images and files', 'Content settings', 'Hosted app data', and 'Media licenses' from the Time range "All time" in Chrome for the original user ("Person") profile under which I first encountered the issue.

Worked like a charm.

I feel like an idiot for not first 'Clearing my browser cache and data' even though it's the first thing I tell others to try when they have browser-related issues. Oh well, live and learn.

Thank you also for including that 'useloom' screen record. It helped me to verify that I was pressing the buttons in the correct order.

Thanks again for all of your help!!
